### PR TITLE
Add documentation for S3 Support

### PR DESCRIPTION
**Pull request by Souf149**
### Changes

Added a new document that describes 

### Issue link

No issue is created. This is an addition for #3191. S3 support has been implemented but there was no documentation yet on how to enable this feature.

### Demo

image

### QA notes

To see the docs I ran `make docs` and then ran `python -m http.server 8021` inside `docs/_build`. 

---

### Additional notes

I have noticed whenever I create a new documentation, this documentation sometimes does not show in my sidebar. It is seemingly random, but it might have to do with my local configuration.

Picture:
image

### CODE REVIEW

File: docs/source/installation-and-deployment/s3-buckets.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+adding 2 more environment variables:
+
+-  **BUCKET_PER_ORG**: This has to be set to ``true`` to tell OpenKAT
+   you want to use a prefix.
+-  **S3_BUCKET_PREFIX**: The name of the prefix you want to use. Make
+   sure that this prefix also follows the `bucket naming rules of
+   AWS <https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html>`__.
+
+After either of these methods have been added files generated by
+``bytes`` should be seen inside the S3 buckets.
```
Comment (ammar92) [inline docs/source/installation-and-deployment/s3-buckets.rst:51]:
```suggestion
Bytes should be seen inside the S3 buckets.
```

File: docs/source/installation-and-deployment/s3-buckets.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+This manual helps you to setup OpenKAT to make use of your S3 buckets.
+
+Enabling S3 buckets for bytes
+=============================
+
+When you want to activate OpenKAT's S3 functionalities you need to have
+an existing S3 service running which is reachable by ``bytes``. When
+this service is up and running, you need to add the following three
+environment variables to the ``.env`` of the machine that is running
+``bytes``
```
Comment (ammar92) [inline docs/source/installation-and-deployment/s3-buckets.rst:16]:
```suggestion
When you want to activate OpenKAT's S3 functionalities you need to have
an existing S3 service running which is reachable by Bytes. When
this service is up and running, you need to add the following three
environment variables to the ``.env`` of the machine that is running
Bytes
```

File: docs/source/installation-and-deployment/s3-buckets.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
+==========
+S3 buckets
+==========
+
+OpenKAT stores most of its data in integrated databases. However, if you
+have access to S3 storage buckets, you may prefer using those instead.
+This manual helps you to setup OpenKAT to make use of your S3 buckets.
+
+Enabling S3 buckets for bytes
```
Comment (ammar92) [inline docs/source/installation-and-deployment/s3-buckets.rst:9]:
```suggestion
Enabling S3 buckets for Bytes
```

File: docs/source/installation-and-deployment/s3-buckets.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+-  **BUCKET_PER_ORG**: This has to be set to ``true`` to tell OpenKAT
+   you want to use a prefix.
+-  **S3_BUCKET_PREFIX**: The name of the prefix you want to use. Make
+   sure that this prefix also follows the `bucket naming rules of
+   AWS <https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html>`__.
+
+After either of these methods have been added files generated by
+``bytes`` should be seen inside the S3 buckets.
+
+Disclaimer
```
Comment (ammar92) [inline docs/source/installation-and-deployment/s3-buckets.rst:53]: You could use the `note` or `warning` directive here if you want it to stand out more, see [this](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-warning) for an example

**Review by ammar92 — COMMENTED** [review]
Great work! My only recommendation is to capitalize *Bytes*, as it refers to our service named *Bytes* and not the data unit. Should be ready to merge soon

I've looked into your issue with the index not being complete, but I wasn't able to reproduce. Perhaps it's a caching issue or something related to the `http.server` module (although unlikely). I usually just open the `index.html` file in the browser directly

File: docs/source/installation-and-deployment/s3-buckets.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+to determine the name of the S3 bucket it should create or use.
+
+S3 bucket names
+===============
+
+OpenKAT offers 2 methods of naming the buckets. The first one is done by
+directly adding the environment variable ``S3_BUCKET`` with a name that
+complies with the `bucket naming rules of
+AWS <https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html>`__.
+This will make OpenKAT throw all of ``bytes``\ ' data into the bucket
```
[the remark this answers is not in the document]
Comment (Souf149) [reply docs/source/installation-and-deployment/s3-buckets.rst:35]: I was wondering on how I should mention it. Thank you ^^

**Review by ammar92 — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Comment by underdarknl** [comment]
Awesome! good work @Souf149

**Pull request merged**